src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Program.cs

```diff
@@ -2,25 +2,137 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+#if (OrganizationalAuth || IndividualB2CAuth)
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Identity.Web;
+using Microsoft.Identity.Web.UI;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Authorization;
+#endif
+using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
+#if (IndividualLocalAuth)
+using Microsoft.AspNetCore.Identity;
+#endif
+#if (OrganizationalAuth)
+using Microsoft.AspNetCore.Mvc.Authorization;
+#endif
+#if (IndividualLocalAuth)
+using Microsoft.EntityFrameworkCore;
+#endif
 using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+#if (MultiOrgAuth)
+using Microsoft.IdentityModel.Tokens;
+#endif
+#if (GenerateGraph)
+using Microsoft.Graph;
+#endif
+#if (IndividualLocalAuth)
+using Company.WebApplication1.Data;
+#endif
 
-namespace Company.WebApplication1
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+#if (IndividualLocalAuth)
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
+#if (UseLocalDB)
+    options.UseSqlServer(
+        builder.Configuration.GetConnectionString("DefaultConnection")));
+#else
+    options.UseSqlite(
+        builder.Configuration.GetConnectionString("DefaultConnection")));
+#endif
+builder.Services.AddDatabaseDeveloperPageExceptionFilter();
+
+builder.Services.AddDefaultIdentity<IdentityUser>(options => options.SignIn.RequireConfirmedAccount = true)
+    .AddEntityFrameworkStores<ApplicationDbContext>();
+#elif (OrganizationalAuth)
+#if (GenerateApiOrGraph)
+var initialScopes = builder.Configuration.GetValue<string>("DownstreamApi:Scopes")?.Split(' ');
+
+#endif
+builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+#if (GenerateApiOrGraph)
+    .AddMicrosoftIdentityWebApp(builder.Configuration.GetSection("AzureAd"))
```

Comment:
Why the nesting of dotted things at the same level?

Reply:
Are you referring to all the Identity methods? If so it's matching what it is today: https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs#L69

Reply:
ugh